src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
-export interface Figure {
 
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+export enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea: () => number,
+}
+
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const longest = [a, b, c];
+    const max: number = Math.max(...longest);
 
+    longest.splice(longest.indexOf(max), 1);
+
+    if ((a <= 0 || b <= 0 || c <= 0) || max >= (longest[0] + longest[1])) {
+      throw new Error('Length less than zero');
+    }
+
+    if (max >= (longest[0] + longest[1])) {
+      throw new Error('The longest side of a triangle'
+      + 'is >= than a sum of two others');
+    }
+  }
+
+  public getArea():number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * ((s - this.a) * (s - this.b) * (s - this.c)));
+
+    return Math.floor(area * 100) / 100;
```

Comment:
you round down to hundredths on each getArea, maybe better create a separate method for it?